lib/OpenQA/Git/ServerAvailability.pm

```diff
@@ -0,0 +1,42 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Git::ServerAvailability;
+use Mojo::Base -strict, -signatures;
+
+use Mojo::File qw(path);
+use Time::Seconds qw(ONE_HOUR);
+use OpenQA::Utils qw(prjdir);
+use Exporter qw(import);
+
+our @EXPORT_OK = qw(report_server_unavailable report_server_available SKIP FAIL);
+
+use constant MAX_DURATION_SECONDS => ($ENV{OPENQA_GIT_SERVER_OUTAGE_DURATION} || (ONE_HOUR / 2));
+
+use constant {
+    SKIP => 'SKIP',
+    FAIL => 'FAIL',
+};
+
+sub _state_file ($app, $server_name) {
+    return path(
+        ($ENV{OPENQA_GIT_SERVER_OUTAGE_FILE} // ($app->config->{global}->{base_dir} // prjdir) . '/git_server_outage')
```

Comment:
Where does `$app->config->{global}->{base_dir}` come from? I can't find such a setting anywhere

Reply:
What does happen if the directory does not exist on the server?

Reply:
oh yes i wanted to remove this, that was a guess on my side, not sure is basedir also a ini setting, if yes whats the easiest way to retrieve it? Changing this to `$ENV{OPENQA_BASEDIR}` for now.

Reply:
My second question is: what happens if this is merged and deployed to o3 and osd and the outage happens, and the `git_server_outage` does not yet exist?

Reply:
fixed earlier using `$file->dirname->make_path;` thanks again for the help :)

Reply:
> What does happen if the directory does not exist on the server?

Why is this resolved if the `make_path` is still not there?